### PR TITLE
fix: only count something as a directory if it has children

### LINKED ISSUE #4515: [BUG]: The count of number of folders when displayed on the project page is incorrect.
### Describe the bug

The display for the number of folders on the main page is incorrect. 

### Steps to Reproduce

1. Boot the application
2. Go to the project selection page
3. Look at the files/folder count
4. See that the folder count is incorrect. 

### Expected Behavior

The folder count should count the number of folders within that directory. 

### Screenshots and Recordings

Image

Default says 20 files / 19 folders

Image
Inside it only has 3 folders. 

### Desktop OS

Ubuntu 22.04 Linux

### Browser

_No response_

### Version

v0.26.5

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by TomPridham**
closes #4515 
Every entry was being counted as a directory on the ProjectCard when it should have only been counting entries with `children`
With this file tree:
```
dsf.kcl
fgk.kcl
folder
folder-1
main.kcl
project.toml
```
before:
Screenshot from 2024-11-23 12-01-30
after:
Screenshot from 2024-11-23 12-01-07

### CODE REVIEW

**Review by jtran — APPROVED** [review]
Looks good to me.

File: src/lib/desktop.ts [hunk 1]
```diff
@@ -307,7 +307,10 @@ const directoryCount = (file: FileEntry) => {
   let count = 0
   if (file.children) {
     for (let entry of file.children) {
-      count += 1
+      // We only want to count FileEntries with children, e.g. folders
+      if (entry.children !== null) {
+        count += 1
+      }
```
Comment (lf94) [inline src/lib/desktop.ts:313]: I would have done `if (!entry.children) continue;` but this works too :P

**Comment by TomPridham** [comment]
Are the test failures expected? They don't seem related to my changes, but I also can't run those failing tests locally.

**Comment by jtran** [comment]
Our CI currently has problems without forks outside the org. I cherry-picked your change on a new PR here: https://github.com/KittyCAD/modeling-app/pull/4595

**Comment by jtran** [comment]
Merged in https://github.com/KittyCAD/modeling-app/pull/4595.

**Pull request closed without merging**